### ISSUE TITLE
Ability to prevent a metric from accumulating during training

### ISSUE THREAD

**Issue by tom-andersson** [tensorflow/tensorflow#50384]
I want a metric that computes the mean value over other metrics. E.g. something like the following:

```
def construct_mean_accuracy(metrics):
    ''' Pass in a list of all other metrics to compute the mean value over all of them. '''

    def mean_accuracy(y_true, y_pred, sample_weight=None):
        return tf.math.reduce_mean([m.result() for m in metrics])

    return mean_accuracy
```

The metrics that I'm computing the mean over here are themselves accumulated (i.e. 'streamed') accuracy values that build up over an epoch. The `mean_accuracy` value printed by TensorFlow during `Model.fit()` is wrong, because TF is automatically accumulating the batch-wise metric values of `mean_accuracy` through a rolling mean (I think). This behaviour has caused confusion for people in the past (see #9498, #15115, #42994), and I couldn't find any documentation on it. Is there a way to prevent the accumulation behaviour and just return the batch-wise value? The desired final `on_epoch_end` value for `mean_accuracy` should just be the mean over the metrics in the final batch (because the other metrics are themselves accumulated metrics).

Let me know if I can make this any clearer.

**Comment by UsharaniPagadala** [comment]
@tom-andersson 

We see that the issue [template](https://github.com/tensorflow/tensorflow/issues/new?assignees=&labels=type%3Afeature&template=30-feature-request.md) has not been filled, could you please do so as it helps us analyse the issue .Thanks

**Comment by jvishnuvardhan** [comment]
@tom-andersson Thanks for creating the issue. `keras` moved to a [new repository](https://github.com/keras-team/keras/issues) and that repo is dedicated for `keras` development. Earlier `keras` team published about the move in TF forum. The link is given below.

https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999

Can you please create a new issue in that repository. When you create issue, can you please share a simple standalone code to reproduce the issue. Standalone code helps to resolve the issue faster. Thanks!
